python/pyspark/tests.py

```diff
@@ -161,6 +161,46 @@ def gen_gs(N, step=1):
             self.assertEqual(k, len(vs))
             self.assertEqual(list(range(k)), list(vs))
 
+    def test_stopiteration_is_raised(self):
+
+        def validate_exception(exc):
+            if isinstance(exc, RuntimeError):
+                self.assertEquals('StopIteration in client code', exc.args[0])
+            else:
+                self.assertIn('StopIteration in client code', exc.java_exception.toString())
```

Comment:
seems we don't need this branch, as we always do `self.assertRaises((Py4JJavaError, RuntimeError))`